automates/text_reading/src/main/scala/org/clulab/aske/automates/apps/ExtractAndAlign.scala

```diff
@@ -866,8 +902,8 @@ object ExtractAndAlign {
     val hypotheses = new ArrayBuffer[ujson.Obj]()
     val linkElKeys = linkElements.keys.toSeq
     // Comment -> Text Var
-    if (linkElKeys.contains(COMMENT) && linkElKeys.contains(TEXT_VAR)) {
-      hypotheses.appendAll(mkLinkHypothesis(linkElements(COMMENT), linkElements(TEXT_VAR), COMMENT_TO_TEXT, alignments(COMMENT_TO_TEXT), debug))
+    if (linkElKeys.contains(COMMENT) && linkElKeys.contains(GLOBAL_VAR)) {
```

Comment:
@maxaalexeeva this function (`getLinkHypotheses`) contains many checks of the following form:
```scala
linkElKeys.contains(<some_key>)
```
The issue here is that we are checking if a key to a map is contained in a list of keys, instead of checking if the key exists in the map. Both of these checks will return identical results, but checking if the key exists in the map will take _O(1)_ time; whereas, checking if the key is in a list of keys will take _O(n)_ time. Luckily the fix is very easy. All of the checks in this function can be changed to the following:
```scala
linkElements.contains(<some_key>)
```
... that check will ensure that the key is in the map and will run in _O(1)_ time as documented [here](https://www.geeksforgeeks.org/scala-map-contains-method-with-example/). Can you please convert all `.contains()` checks in this function using the pattern above?

Additionally, the following check is run **many** times in this function:
```scala
linkElKeys.contains(GLOBAL_VAR)
```
Ideally, we only want to run this check once. Can you perform the `.contains()` check for _Global Var_ in an outer if-statement and then perform all subsequent `.contains()` checks for elements that link to _Global Var_ under that if-statement? That will greatly reduce the number of redundant checks happening in this function.